kyuubi-rest-client/src/main/java/org/apache/kyuubi/client/HttpClientFactory.java

```diff
@@ -35,23 +42,35 @@ public class HttpClientFactory {
   public static CloseableHttpClient createHttpClient(RestClientConf conf) {
     RequestConfig requestConfig =
         RequestConfig.custom()
-            .setSocketTimeout(conf.getSocketTimeout())
-            .setConnectTimeout(conf.getConnectTimeout())
+            .setResponseTimeout(conf.getSocketTimeout(), TimeUnit.MILLISECONDS)
             .build();
+    Registry<ConnectionSocketFactory> socketFactoryRegistry;
     SSLConnectionSocketFactory sslSocketFactory;
     try {
       TrustStrategy acceptingTrustStrategy = (cert, authType) -> true;
       SSLContext sslContext =
           SSLContexts.custom().loadTrustMaterial(null, acceptingTrustStrategy).build();
       sslSocketFactory = new SSLConnectionSocketFactory(sslContext, NoopHostnameVerifier.INSTANCE);
+      socketFactoryRegistry =
+          RegistryBuilder.<ConnectionSocketFactory>create()
+              .register("https", sslSocketFactory)
+              .register("http", new PlainConnectionSocketFactory())
+              .build();
     } catch (Exception e) {
       LOG.error("Error: ", e);
       throw new KyuubiRestException("Failed to create HttpClient", e);
     }
+    BasicHttpClientConnectionManager cm =
+        new BasicHttpClientConnectionManager(socketFactoryRegistry);
+    ConnectionConfig connConfig =
+        ConnectionConfig.custom()
+            .setConnectTimeout(conf.getConnectTimeout(), TimeUnit.MILLISECONDS)
+            .build();
+    cm.setConnectionConfig(connConfig);
 
     return HttpClientBuilder.create()
         .setDefaultRequestConfig(requestConfig)
-        .setSSLSocketFactory(sslSocketFactory)
+        .setConnectionManager(cm)
```

Comment:
note: `setSSLSocketFactory` method is removed in httpclient5, migrated to set socketFactoryRegistry via the ConnectionManager.